### ISSUE TITLE
Terraform creating Network Interfaces out of order (libvirt)

### ISSUE THREAD

**Issue by nathanjmartin** [hashicorp/terraform#21921]
Hi there,

I'm trying to use Terraform to generate a VyOS virtual machine that has 2 interfaces, one pointing to an outside network (to other VyOS routers), and another pointing to the router's inside network (to centos virtual machines).
The problem I am having is, terraform seems to inconsistent when it comes to the order of the network interfaces it creates.

I'm using libvirt_network, libvirt_domain, etc for my current project. 
I know there is a depend_on function but that only seems to work at the resource level, however I'm needing something that makes one resource's network interface wait (the inner one) until the outer interface is created first (so the order in vyos is consistent with eth1, eth2).
I am not sure if I can just not find the correct documentation on this, or if the capability just does not exist.

Here is my code that I am having issues with:

```
resource "libvirt_volume" "router_1" {
  name = "router_1.qcow2"
  pool = "default"
  source = "/home/nmartin/projects/VMs/vyos_router.qcow2"
}

resource "libvirt_domain" "router_1" {
  name   = "router_1"
  memory = "2048"
  vcpu   = 2

  network_interface {
    network_id = "${libvirt_network.router_net.id}"
    network_name = "router_net"
    mac = "AA:BB:CC:11:22:01"

  }

  network_interface {
    network_id = "${libvirt_network.inside_1.id}"
    network_name = "inside_1"
    mac = "AA:BB:CC:11:23:01"
  }

  disk {
    volume_id = "${libvirt_volume.router_1.id}"
  }

  console {
    type = "pty"
    target_type = "serial"
    target_port = "0"
  }

  graphics {
    type = "spice"
    listen_type = "address"
    autoport = true
  }
```

So basically I am wanting terraform to make the second network interface wait until the first one is created (router_net)

To help picture this (can't take a picture bc of policies)
When I do an ip addr on the VyOS machine
I sometimes get the following:

1. lo
2. eth2
3. eth1

I need the order to consistently be (as sometimes it is correct, but with multiple machines being made, plenty of them have the incorrect order)

1. lo
2. eth1
3. eth2

NOT
1. lo
2. eth2
3. eth1

**Comment by nathanjmartin** [comment]
I'm not sure why the MD stuff didn't work, if someone could fix it'd be greatly appreciated.
Also, I'm using Terraform v0.11.14

**Comment by jbardin** [comment]
Hi @nathanjmartin,

Resource are created an a whole as far as terraform, so any ordering of the network interfaces needs to be determined by the `libvirt_domain` resource. If the order does matter, you would need to file an issue with the libvirt provider maintainers.
